src/pybind/mgr/dashboard/frontend/src/app/shared/validators/cd-validators.ts

```diff
@@ -56,4 +56,44 @@ export class CdValidators {
       return success ? { required: true } : null;
     };
   }
+
+  /**
+   * Custom validation by passing a name for the error and a function as error condition.
+   *
+   * @param {string} error
+   * @param {Function} condition - a truthy return value will trigger the error
+   * @returns {ValidatorFn}
+   */
+  static custom(error: string, condition: Function): ValidatorFn {
+    return (control: AbstractControl): { [key: string]: any } => {
+      const value = condition.call(this, control.value);
+      if (value) {
+        return { [error]: value };
+      }
+      return null;
+    };
+  }
+
+  /**
+   * Only validate form control if condition is true with validators.
+   *
+   * @param {AbstractControl} formControl
+   * @param {Function} condition
+   * @param {ValidatorFn[]} validators
+   */
+  static onlyValidate(
```

Comment:
I suggest to name it ``validateIf`` which in my opinion says more clearly what this validator is doing. Additionally this name is more equal to the already existing ``requiredIf`` validator.

Reply:
Funny fact - that was my first name for this function ;)